midterm_project/sorin_adrian/shoppingList.py

```diff
@@ -0,0 +1,294 @@
+
+    def __getitem__(self, index):
+        return self._recipesbox_list[index]
+
+    def __contains__(self, item):
+        return item in self._recipesbox_list
+
+    def __setitem__(self, index, value):
+        self._recipesbox_list[index] = value
+
+    def __delitem__(self, index):
+        del self._recipesbox_list[index]
+
+    def remove(self, item):
+        self._recipesbox_list.remove(item)
+
+    def append(self, item):
+        self._recipesbox_list.append(item)
+
+    # eliminate a recipe from recipes_box list by its index
+    def pop(self, index=None):
+        if index:
+            return self._recipesbox_list.pop(index)
+        else:
+            return self._recipesbox_list.pop()
+
+    # method get a recipe as argument, extract the recipe from recipes_box and print it
+    def pick(self, recipe=None):
+        if recipe:
+            index = self._recipesbox_list.index(recipe)
+        else:
+            max_rand_no = len(self._recipesbox_list)
+            index = randrange(0, max_rand_no, 1)
+
+        return self._recipesbox_list[index]
+
+
+# class create a instance for archive ingredients for recipes
+class Fridge(PrettyPrinterMixin):
+
+    def __init__(self):
+        self._fridge = {}
+
+    def __getitem__(self, item):
+        return self._fridge[item]
+
+    def __iter__(self):
+        return iter(self._fridge)
+
+    def __len__(self):
+        return len(self._fridge)
+
+    def __contains__(self, item):
+        return item in self._fridge.keys()
+
+    def __delitem__(self, key):
+        del self._fridge[key]
+
+    def keys(self):
+        return self._fridge.keys()
+
+    def values(self):
+        return self._fridge.values()
+
+    def items(self):
+        return self._fridge.items()
+
+    def update(self, *args, **kwargs):
+        self._fridge.update(*args, **kwargs)
+
+    # methode allow to update the quantity of one existing ingredient from the fridge
+    def update_quantity(self, ingredient, quantity):
+        updated_quantity = self._fridge[ingredient] + quantity
+        if updated_quantity > 0:
+            self._fridge[ingredient] = updated_quantity
+        else:
+            print(f'There is no more \033[31m\033[01m{ingredient}\033[0m in the fridge, go shopping !!!\n')
+            del self._fridge[ingredient]
+
+    # methode check if there are in the frig all the required ingredients for a recipe
+    def check_recipe(self, recipe):
+        recipe_ingredients = set(recipe.recipe_ingredients)
+        fridge_ingredients = set(self._fridge.keys())
+        ingredients_in = recipe_ingredients.intersection(fridge_ingredients)
+        ingredients_off = recipe_ingredients.difference(fridge_ingredients)
+
+        message = f'There are no missing ingredients for preparing the {recipe.recipe_name}'
+        message_2 = ['No ingredients in fridge']
+
+        return ingredients_in if ingredients_in else message_2, ingredients_off if ingredients_off else message
```

Comment:
this is a bit too complicated can you add the statements in two separat variabels and then return them :smile: ?